### PR TITLE
Pylint fixes

### PR DESCRIPTION
**Pull request by ricardobranco777**
This PR adds a Makefile to run `make test` and contains some fixes for low-hanging fruit from the output of Pylint & Flake8.

The Makefile notes the other fixes that must be made.

There are other Pythonic changes that can be done but I'll leave those for later after resolving all Pylint & Flake8 warnings.

### CODE REVIEW

File: ansible/roles/sap_hana_install/tests/install/run-sap_hana_install-install-tests.py [hunk 1]
```diff
@@ -4,34 +4,35 @@
 import sys
```
Comment (mpagot) [inline ansible/roles/sap_hana_install/tests/install/run-sap_hana_install-install-tests.py:4]: This is not one of the files of the glue script, does not mean that fixes has not to be accepted, but I'm not sure what exactly this script is about. So I'm not sure that we have to apply the same linting rules here too

File: ansible/roles/sap_hana_install/tests/sapcar/run-sap_hana_install-sapcar-tests.py [hunk 2]
```diff
@@ -9,13 +9,13 @@
 # output field delimiter for displaying the results:
 _field_delimiter = '\t'
```
Comment (mpagot) [inline ansible/roles/sap_hana_install/tests/sapcar/run-sap_hana_install-sapcar-tests.py:10]: as before

File: scripts/qesap/test/unit/test_qesap_terraform.py [hunk 3]
```diff
@@ -55,7 +59,7 @@ def test_terraform_terraform_logs(run, terraform_cmd_args, args_helper, config_y
     provider = 'mangiafuoco'
     conf = config_yaml_sample(provider)
 
-    args, terraform_dir, _, _ = args_helper(provider, conf, '')
+    args, *_ = args_helper(provider, conf, '')
```
Comment (mpagot) [inline scripts/qesap/test/unit/test_qesap_terraform.py:62]: what is `*_` ?

File: scripts/qesap/test/unit/test_subprocess_run.py [hunk 4]
```diff
@@ -29,10 +29,10 @@ def test_multilines():
     test_text = ''
     for s in str_list:
         test_text += (s*10) + "\n"
-    exit_code, stdout_list = subprocess_run(['echo', test_text])
+    _, stdout_list = subprocess_run(['echo', test_text])
     assert len(stdout_list) == len(str_list)+1
-    for i in range(len(str_list)):
-        assert str_list[i]*10 == stdout_list[i].strip()
+    for _, x in enumerate(str_list):
```
Comment (mpagot) [inline scripts/qesap/test/unit/test_subprocess_run.py:34]: like

File: ansible/roles/sap_hana_install/tests/install/run-sap_hana_install-install-tests.py (hunk 1 above)
Comment (ricardobranco777) [reply ansible/roles/sap_hana_install/tests/install/run-sap_hana_install-install-tests.py:4]: > This is not one of the files of the glue script, does not mean that fixes has not to be accepted, but I'm not sure what exactly this script is about. So I'm not sure that we have to apply the same linting rules here too

This is an unused import.

File: scripts/qesap/test/unit/test_qesap_terraform.py (hunk 3 above)
Comment (ricardobranco777) [reply scripts/qesap/test/unit/test_qesap_terraform.py:62]: `*foobar` catches the rest of the returned values in an array.  `_` is just the throaway variable name by convention.

**Review by mpagot — APPROVED** [review]
LGTM

File: scripts/qesap/test/unit/test_qesap_configure.py [hunk 5]
```diff
@@ -310,9 +310,9 @@ def test_configure_tfvarstemplate(base_args, tmpdir):
     is missing
     """
     provider = 'pinocchio'
-    os.makedirs(os.path.join(tmpdir,'terraform', provider))
+    os.makedirs(os.path.join(tmpdir, 'terraform', provider))
     config_file_name = str(tmpdir / 'config.yaml')
-    with open(config_file_name, 'w') as file:
+    with open(config_file_name, 'w', encoding="utf-8", encoding="utf-8") as file:
```
Comment (mpagot) [inline scripts/qesap/test/unit/test_qesap_configure.py]: E     File "/home/runner/work/qe-sap-deployment/qe-sap-deployment/scripts/qesap/test/unit/test_qesap_configure.py", line 315
  E       with open(config_file_name, 'w', encoding="utf-8", encoding="utf-8") as file:

**Review by mpagot — APPROVED** [review]
LGTM

File: scripts/qesap/test/unit/test_subprocess_run.py (hunk 4 above)
Comment (mpagot) [reply scripts/qesap/test/unit/test_subprocess_run.py:34]: https://github.com/SUSE/qe-sap-deployment/runs/8129579939?check_suite_focus=true

This change result in a unit test failure
```
          for _, x in enumerate(str_list):
  >           assert x*10 == x.strip()
  E           AssertionError: assert 'aaaaaaaaaa' == 'a'
  E             - a
  E             + aaaaaaaaaa
```

**Pull request merged**